tests/Unit/NumericalAlgorithms/Interpolation/Test_RegularGridInterpolant.cpp

```diff
@@ -0,0 +1,299 @@
+      for (size_t d = 0; d < Dim; ++d) {
+        result.get(d)[iter.collapsed_index()] =
+            gsl::at(target_1d_logical_coords, d)[iter()[d]];
+      }
+    }
+    return result;
+  };
+
+  for (size_t dim_to_override = 0; dim_to_override < Dim; ++dim_to_override) {
+    auto target_logical_coords =
+        make_target_logical_coordinates(dim_to_override);
+    const auto target_coords = map(target_logical_coords);
+
+    // Set up variables
+    using tags = tmpl::list<TestTags::ScalarTag<Dim>, TestTags::Vector<Dim>>;
+    Variables<tags> source_vars(source_mesh.number_of_grid_points());
+    Variables<tags> expected_result(get<0>(target_coords).size());
+
+    // Set up interpolator
+    auto override_target_mesh_with_1d_coords = make_array<Dim>(DataVector{});
+    gsl::at(override_target_mesh_with_1d_coords, dim_to_override) =
+        override_coords;
+    const intrp::RegularGrid<Dim> regular_grid_interpolant(
+        source_mesh, target_mesh, override_target_mesh_with_1d_coords);
+
+    // Test only the highest-order polynomial x^a y^b z^c on the source mesh.
+    std::array<std::unique_ptr<MathFunction<1>>, Dim> functions;
+    for (size_t d = 0; d < Dim; ++d) {
+      gsl::at(functions, d) =
+          std::make_unique<MathFunctions::PowX>(source_mesh.extents(d) - 1);
+    }
+    MathFunctions::TensorProduct<Dim> f(1.0, std::move(functions));
+
+    // Fill source and expected destination Variables with analytic solution.
+    tmpl::for_each<tags>([
+      &f, &source_coords, &target_coords, &source_vars, &expected_result
+    ](auto tag) noexcept {
+      using Tag = tmpl::type_from<decltype(tag)>;
+      get<Tag>(source_vars) = Tag::fill_values(f, source_coords);
+      get<Tag>(expected_result) = Tag::fill_values(f, target_coords);
+    });
+
+    // Interpolate
+    // (g++ 7.2.0 does not allow `const auto result` here)
+    const Variables<tags> result =
+        regular_grid_interpolant.interpolate(source_vars);
+
+    // For 3D extrapolation, errors get large-ish.
+    Approx local_approx = Approx::custom().epsilon(1.e-9).scale(1.0);
+    tmpl::for_each<tags>(
+        [&result, &expected_result, &local_approx ](auto tag) noexcept {
+          using Tag = tmpl::type_from<decltype(tag)>;
+          CHECK_ITERABLE_CUSTOM_APPROX(get<Tag>(result),
+                                       get<Tag>(expected_result), local_approx);
+        });
+  }
+}
+
+void test_1d_regular_interpolation() {
+  const size_t start_points = 4;
+  const size_t end_points = 6;
+  const DataVector coords = {1.0, 1.5, 2.5, 3.0};  // some extrapolating points
+  for (size_t n = start_points; n < end_points; ++n) {
+    const auto mesh_lgl = Mesh<1>{n, Spectral::Basis::Legendre,
+                                  Spectral::Quadrature::GaussLobatto};
+    const auto mesh_lg_high_res =
+        Mesh<1>{n + 2, Spectral::Basis::Legendre, Spectral::Quadrature::Gauss};
+    test_regular_interpolation(mesh_lgl, mesh_lg_high_res);
```

Comment:
Test cases where the meshes agree in at least one dimension.

Reply:
Surprise, adding this test revealed a bug. Thanks for catching this :+1: